app/code/Magento/Swatches/Model/Plugin/FilterRenderer.php

```diff
@@ -3,63 +3,82 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
+
 namespace Magento\Swatches\Model\Plugin;
 
-/**
- * Class FilterRenderer
- */
+use Closure;
+use Magento\Catalog\Model\Layer\Filter\FilterInterface;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\View\LayoutInterface;
+use Magento\LayeredNavigation\Block\Navigation\FilterRenderer as Subject;
+use Magento\Swatches\Block\LayeredNavigation\RenderLayered;
+use Magento\Swatches\Helper\Data;
+use Magento\Swatches\ViewModel\Product\Renderer\Configurable as ConfigurableViewModel;
+
 class FilterRenderer
 {
     /**
-     * @var \Magento\Framework\View\LayoutInterface
+     * @var LayoutInterface
      */
     protected $layout;
 
     /**
-     * Path to RenderLayered Block
-     *
-     * @var string
+     * @var Data
      */
-    protected $block = \Magento\Swatches\Block\LayeredNavigation\RenderLayered::class;
+    protected $swatchHelper;
 
     /**
-     * @var \Magento\Swatches\Helper\Data
+     * @var ConfigurableViewModel|null
      */
-    protected $swatchHelper;
+    private $configurableViewModel;
 
     /**
-     * @param \Magento\Framework\View\LayoutInterface $layout
-     * @param \Magento\Swatches\Helper\Data $swatchHelper
+     * @var string
+     */
+    protected $block = RenderLayered::class;
+
+    /**
+     * @param LayoutInterface $layout
+     * @param Data $swatchHelper
+     * @param ConfigurableViewModel|null $configurableViewModel
      */
     public function __construct(
-        \Magento\Framework\View\LayoutInterface $layout,
-        \Magento\Swatches\Helper\Data $swatchHelper
+        LayoutInterface $layout,
+        Data $swatchHelper,
+        ?ConfigurableViewModel $configurableViewModel = null
     ) {
         $this->layout = $layout;
         $this->swatchHelper = $swatchHelper;
+        $this->configurableViewModel = $configurableViewModel
```

Comment:
You may not follow BC development rules in plugins.

Reply:
Done